### PR TITLE
[web] Fix `kIsWeb` for Dart2wasm.

### PR DESCRIPTION
**Pull request by joshualitt**

### CODE REVIEW

File: packages/flutter/lib/src/foundation/constants.dart [hunk 1]
```diff
@@ -68,10 +68,5 @@ const bool kDebugMode = !kReleaseMode && !kProfileMode;
...
 /// A constant that is true if the application was compiled to run on the web.
-///
-/// This implementation takes advantage of the fact that JavaScript does not
-/// support integers. In this environment, Dart's doubles and ints are
-/// backed by the same kind of object. Thus a double `0.0` is identical
-/// to an integer `0`. This is not true for Dart code running in AOT or on the
-/// VM.
-const bool kIsWeb = identical(0, 0.0);
+/// ignore: unnecessary_const
+const bool kIsWeb = !const bool.fromEnvironment('dart.library.io');
```
Comment (jonahwilliams) [inline packages/flutter/lib/src/foundation/constants.dart:72]: Should be the same as?

```suggestion
const bool kIsWeb = !bool.fromEnvironment('dart.library.io');
```

**Review by jonahwilliams — APPROVED** [review]
Rest in peace, my hacky friend.

LGTM

File: packages/flutter/lib/src/widgets/platform_selectable_region_context_menu.dart [hunk 2]
```diff
@@ -2,4 +2,5 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-export '_platform_selectable_region_context_menu_io.dart' if(dart.library.html) '_platform_selectable_region_context_menu_web.dart';
+export '_platform_selectable_region_context_menu_io.dart'
+if(dart.library.js_util) '_platform_selectable_region_context_menu_web.dart';
```
Comment (justinmc) [inline packages/flutter/lib/src/widgets/platform_selectable_region_context_menu.dart:6]: Nit: Space in between? `if (`

**Review by justinmc — APPROVED** [review]
LGTM 👍 

I'm somehow sad to see this go!

**Comment by joshualitt** [comment]
Just a heads up, this CL is waiting for an analyzer roll, and will probably fail until those changes land.

**Pull request merged**